lib/CleantalkSP/SpbctWP/RemoteCalls.php

```diff
@@ -176,31 +176,38 @@ public static function action__download__quarantine_file() // phpcs:ignore PSR1.
      */
     public static function action__update_settings() // phpcs:ignore PSR1.Methods.CamelCapsMethodName.NotCamelCaps
     {
+        if ( ! headers_sent() ) {
+            header('Content-Type: application/json');
+        }
 
         global $spbc;
 
-        foreach ( $spbc->default_settings as $setting ) {
-            if ( Request::get($setting) ) {
-                $var = Request::get($setting);
-                $type = gettype($spbc->settings[$setting]);
+        $incoming_settings = json_decode(Request::get('settings'), true);
 
-                switch ( $type ) {
-                    case 'string':
-                        $var = preg_replace(array('/=/', '/`/'), '', $var);
-                        break;
-                    case 'array':
-                        $var = explode(',', $var);
-                        break;
-                }
+        if ( ! $incoming_settings ) {
+            die(json_encode(['error' => 'No settings provided']));
```

Comment:
WA wp_send_json() instead of die(json_encode) ?

Reply:
https://github.com/CleanTalk/security-malware-firewall/commit/c8fcd46c05a75c13423ffe7e2f8eff6be2109497